### PR TITLE
add  kubesphere

### PR DESCRIPTION
**Pull request by misteruly**
Signed-off-by: misteruly <lanpeng@yunify.com>

## BACKGROUND
* Add information about the background and reason for the change.
Add KubeSphere app to marketplace
-----------------------------------------------------------------------

## Changes
* List the changes that you made
Add KubeSphere app to marketplace
-----------------------------------------------------------------------

## Checklist
- [ ] review the [contributing doc](https://github.com/digitalocean/marketplace-kubernetes/blob/master/CONTRIBUTING.md) with steps for both adding or updating your application (if applicable)
------------------------------------------------------------------------

Reviewer: @marketplace-eng

### CODE REVIEW

**Comment by FeynmanZhou** [comment]
@keladhruv @marketplace-eng Hi team, could you please help to review this PR? Thanks!

**Comment by jgannondo** [comment]
@FeynmanZhou 👋 hi - thanks for checking in. We are on a company holiday today and yesterday, I'm going to ping in Slack to make sure someone can take a look early next week.

File: stacks/kubesphere/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+
+set -e
+
+################################################################################
+# repo
+################################################################################
+helm repo add test https://charts.kubesphere.io/test
```
Comment (keladhruv) [inline stacks/kubesphere/deploy.sh:8]: To confirm, is the name of the helm repo test in production as well?

File: stacks/kubesphere/values.yaml [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+      cloudhubQuicNodePort: "30001"
+      cloudhubHttpsNodePort: "30002"
+      cloudstreamNodePort: "30003"
+      tunnelNodePort: "30004"
+  edgeWatcher:
+    nodeSelector: {"node-role.kubernetes.io/worker": ""}
+    tolerations: []
+    edgeWatcherAgent:
+      nodeSelector: {"node-role.kubernetes.io/worker": ""}
+      tolerations: []
```
Comment (keladhruv) [inline stacks/kubesphere/values.yaml:197]: Nit: Extra line

**Comment by keladhruv** [comment]
The files are currently not executable, please changes the file permissions.

**Review by keladhruv — CHANGES_REQUESTED** [review]
Added comments.

File: stacks/kubesphere/deploy.sh (hunk 1 above)
Comment (misteruly) [reply stacks/kubesphere/deploy.sh:8]: > The files are currently not executable, please changes the file permissions.
Execute permissions have been added

Comment (FeynmanZhou) [reply stacks/kubesphere/deploy.sh:8]: Yes, but this repo name and  URL may mislead the novices. We will replace this URL with `main` and `https://charts.kubesphere.io/test`.

Comment (misteruly) [reply stacks/kubesphere/deploy.sh]: This has been changed to the main library  @keladhruv  @FeynmanZhou

**Comment by FeynmanZhou** [comment]
Fixed and updated.

/cc @keladhruv

**Review by keladhruv — APPROVED** [review]
LGTM

**Comment by jgannondo** [comment]
@FeynmanZhou please make sure you take this PR and create your app listing in marketplace.digitalocean.com/vendorportal. This last step is needed to make your app live to all of our customers in the Marketplace. Thanks!

**Pull request merged**